src/develop/blend.c

```diff
@@ -2426,6 +2503,331 @@ static void _blend_HSV_color(const _blend_buffer_desc_t *bd, const float *a, flo
+      b[j + 3] = local_opacity;
+    }
+  }
+  else if(bd->cst == iop_cs_rgb)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
+
+      b[j + 0] = (a[j + 0] * (1.0f - local_opacity)) + b[j + 0] * local_opacity;
+      b[j + 1] = a[j + 1];
+      b[j + 2] = a[j + 2];
+      b[j + 3] = local_opacity;
+    }
+  }
+  else /* if(bd->cst == iop_cs_RAW) */
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      for(int k = 0; k < bd->bch; k++) b[j + k] = a[j + k]; // Noop for RAW (unclamped)
+    }
+  }
+}
+
+/* blend only R-channel in RGB color space without any clamping (a noop for
+ * other color spaces) */
+static void _blend_RGB_G(const _blend_buffer_desc_t *bd, const float *a, float *b, const float *mask,
+                         int flag)
+{
+  float max[4] = { 0 }, min[4] = { 0 };
+  _blend_colorspace_channel_range(bd->cst, min, max);
+
+  if(bd->cst == iop_cs_Lab)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
+      for(int k = 0; k < bd->bch; k++) b[j + k] = a[j + k]; // Noop for Lab (unclamped)
+      b[j + 3] = local_opacity;
+    }
+  }
+  else if(bd->cst == iop_cs_rgb)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
+
+      b[j + 0] = a[j + 0];
+      b[j + 1] = (a[j + 1] * (1.0f - local_opacity)) + b[j + 1] * local_opacity;
+      b[j + 2] = a[j + 2];
+      b[j + 3] = local_opacity;
+    }
+  }
+  else /* if(bd->cst == iop_cs_RAW) */
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      for(int k = 0; k < bd->bch; k++) b[j + k] = a[j + k]; // Noop for RAW (unclamped)
+    }
+  }
+}
+
+/* blend only R-channel in RGB color space without any clamping (a noop for
+ * other color spaces) */
+static void _blend_RGB_B(const _blend_buffer_desc_t *bd, const float *a, float *b, const float *mask,
+                         int flag)
+{
+  float max[4] = { 0 }, min[4] = { 0 };
+  _blend_colorspace_channel_range(bd->cst, min, max);
+
+  if(bd->cst == iop_cs_Lab)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
+      for(int k = 0; k < bd->bch; k++) b[j + k] = a[j + k]; // Noop for Lab (unclamped)
+      b[j + 3] = local_opacity;
+    }
+  }
+  else if(bd->cst == iop_cs_rgb)
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      float local_opacity = mask[i];
+
+      b[j + 0] = a[j + 0];
+      b[j + 1] = a[j + 1];
+      b[j + 2] = (a[j + 2] * (1.0f - local_opacity)) + b[j + 2] * local_opacity;
+      b[j + 3] = local_opacity;
+    }
+  }
+  else /* if(bd->cst == iop_cs_RAW) */
+  {
+    for(size_t i = 0, j = 0; j < bd->stride; i++, j += bd->ch)
+    {
+      for(int k = 0; k < bd->bch; k++) b[j + k] = a[j + k]; // Noop for RAW (unclamped)
```

Comment:
This exact block is added in 5 different places even in just this pr,
Are you seriously telling me that 6*4 extra lines of duplication is that much better than a call?